### ISSUE TITLE
Sleep by number of tracks instead of minutes

### ISSUE THREAD

**Issue by phly95** [PaulWoitaschek/Voice#729]
When doing language courses, they don't always tell you when the track ends or begins (eg. Michel Thomas Method)

It would be nice if the sleep timer could be set by setting say 3 tracks, and when it stops, then you know you've done enough learning for now.

Sure you could have a sleep timer, but then it'll stop in the middle of the lesson.

This could be implemented using a single switch that changes the units from time to tracks.

**Comment by phly95** [comment]
This could also help people who wish to have the feature to sleep when a chapter is over (see #464) while keeping the UI pretty much intact and avoiding making it cluttered with options.

**Comment by shadowkity** [comment]
I also agree with this, I have made a similar request for this feature too [here](https://github.com/PaulWoitaschek/Voice/issues/712). For me, it's more useful to auto-stop per track/chapter and I think more people can benefit with this too.

**Comment by nimbus77** [comment]
I would also like to have this feature.

**Comment by phly95** [comment]
I think the easiest way to keep it from cluttering the sleep menu is to replace "min" with a drop-down menu. So it would look like

20 |min⬇️|
    [tracks  ]
    [hours   ]
    [seconds]
    [days     ]

Under a drop-down menu, it takes up no additional space unless you tap on it and choose, therefore, no additional clutter.

**Comment by nimbus77** [comment]
That sounds perfectly alright to me.

**Comment by RayBB** [comment]
I also think this would be a really great feature. It's especially nice for books that don't indicate when a new chapter begins.
